### DISCUSSION TITLE
dosbox.exe -v and -V are case sensitive

### DISCUSSION THREAD

**Discussion by Torinde** [dosbox-staging/dosbox-staging#4030] [General]
`dosbox.exe -v`
![image](https://github.com/user-attachments/assets/0da18dc8-102c-4e6a-9787-70bc2c4472e3)

`dosbox.exe -V`
![image](https://github.com/user-attachments/assets/26fd7e39-80fc-4599-b094-0c898415d737)

Related:
- #4690

**Comment by interloper98** [comment]
Looks like the loguru lib code is parsing `-v` as a verbosity level (INFO / WARNING / ERROR) before the argv gets into the sdlmain code (but when it does, the `-v` is re-parsed by DOSBox Staging as a version request). 

I'm not sure what we should do (or what's possible):
 - make loguru not parse argv? 
 - make loguru "eat" the `-v LOGLEVEL` portion before argv hits the DOSBox Staging code?).

**Reply by johnnovak** [reply]
@Torinde @interloper98 

Dunno, why would we need to do anything? Case sensitive options are fine, and what we have is fine, too:

<img width="549" alt="image" src="https://github.com/user-attachments/assets/f034ed6b-c9b1-4381-a1c0-32621e89f59c">

Here is another common tool with case sensitive options:

<img width="704" alt="image" src="https://github.com/user-attachments/assets/6effadcb-4a67-4ba7-9ef9-94aed55afae0">

Then from the help of `tar`:

<img width="737" alt="image" src="https://github.com/user-attachments/assets/6c1297f0-60cb-4ccf-bc1d-52b04ae2d350">

**Reply by Torinde** [reply]
True that the help is literally correct (V vs v), I just found it strange that most (all other?) settings are lower case, but exactly the one I used gave me an error.

Do nothing is valid option, yes. And I can't judge what the implications/complications of working around that strange hurdle are (what's so special about lowercase v?).

**Reply by shermp** [reply]
Another option is to change the default flag in Loguru to something other than `-v` by setting the `verbosity_flag` option.

**Comment by interloper98** [comment]
The issue that @Torinde's hitting is that Loguru (itself, not Staging) is intercepting and parsing `argv` where it sees the lower-case `-v` and it complains about it (`"missing verbosity level"` in the first screenshot above), because loguru's `-v` expects it to be `-v LEVEL`.

> why would we need to do anything? Case sensitive options are fine

That would be fine if it were true (DOSBox Staging says that capital `-V` is version, so if it were true we would ignore the lowercase `-v`).

But DOSBox Staging's CLI is _not_ case sensitive. `command_line.cpp` is using `strncmp`, `strncasecmp`, and `strncmp`. Try this: `dosbox.exe -v INFO`:
- Loguru parses its verbosity level, *and*
- DOSBox Staging treats the `-v` as `-V`, so it prints the version and then quits.

@johnnovak , I guess the fix is to make the CLI key names case-sensitive?

**Reply by Torinde** [reply]
Don't know if it matters, but just to clarify (in case it's not obvious from above) - I was doing `dosbox.exe -v` in CMD.exe, not in DOSBox shell itself.

**Reply by Torinde** [reply]
Now I tried running `dosbox.exe -v` inside DOSbox shell (another .exe from another folder, not recursively the same one) - result was blinking cursor and not working X close button (X close on the console window worked) :)

Would've been nice to get instead the regular stub message "This program cannot be run in DOS mode."

**Reply by johnnovak** [reply]
> I guess the fix is to make the CLI key names case-sensitive?

@interloper98 Yeah ok, I'm happy with tightening the command line param checking like that.

Very minor thing, though. 99% of people would do as instructed by the help and never try whether the options are case insensitive or not... Well, I never try that with any CLI tool I use -- half of them might be lenient like DOSBox currently for all I know 😆
